book/src/libs/wasp/components/generator.md

```diff
@@ -0,0 +1,57 @@
+# WASP - Generator
+
+A **Generator** is a component that encapsulates all the characteristics of load generation, including:
+* Load type:
+  * RPS (requests per second)
+  * VUs (virtual users)
+* [Load schedule](./schedule.md)
+* Call logic
+* Response data
+* [Sampling](./sampler.md)
+* Timeouts
+
+> [!WARNING]  
+> RPS load type can only be used with a `Gun`, while VUs can only be used with a `VirtualUser`.
+
+---
+
+### Choosing Between `Gun` and `VirtualUser`
+
+#### **`Gun`**
+* Best for **stateless protocols** (e.g., HTTP).
+* Simplistic in nature; ideal for executing a single operation that does not require setup or teardown.
+* Operates using a **closed model**, where:
```

Comment:
It is an `open` model, if the service can't process our `N RPS` we'll use more goroutines to test that it can scale strictly to our required workload of `N RPS`
Like `Google`, it is `open` no one can prohibit you to use `Google`